src/Arcus.BackgroundJobs.DataBricks/QueryRepeatedlyDatabricksReportJob.cs

```diff
@@ -0,0 +1,92 @@
+        /// <param name="configuration">Configuration of the application</param>
+        /// <param name="serviceProvider">Collection of services that are configured</param>
+        /// <param name="logger">Logger to write telemetry to</param>
+        public QueryRepeatedlyDatabricksReportJob(
+            QueryRepeatedlyDatabricksReportJobOptions options,
+            IConfiguration configuration,
+            IServiceProvider serviceProvider,
+            ILogger logger) : base(configuration, serviceProvider, logger)
+        {
+            Guard.NotNull(options, nameof(options));
+
+            _options = options;
+        }
+
+        /// <summary>
+        /// This method is called when the <see cref="T:Microsoft.Extensions.Hosting.IHostedService" /> starts. The implementation should return a task that represents
+        /// the lifetime of the long running operation(s) being performed.
+        /// </summary>
+        /// <param name="stoppingToken">Triggered when <see cref="M:Microsoft.Extensions.Hosting.IHostedService.StopAsync(System.Threading.CancellationToken)" /> is called.</param>
+        /// <returns>A <see cref="T:System.Threading.Tasks.Task" /> that represents the long running operations.</returns>
+        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+        {
+            var secretProvider = ServiceProvider.GetRequiredService<ISecretProvider>();
+
+            using (DatabricksClient client = await _options.CreateDatabricksClientAsync(secretProvider))
+            using (var databricksInfoProvider = new DatabricksInfoProvider(client))
+            {
+                DateTimeOffset last = DateTime.UtcNow;
+                while (!stoppingToken.IsCancellationRequested)
+                {
+                    DateTimeOffset next = DateTimeOffset.UtcNow;
+                    Logger.LogInformation(
+                        "Job monitor for Databricks is starting at {TriggerTime} for time windows {WindowStart} - {WindowEnd}",
+                        DateTimeOffset.UtcNow, last, next);
+
+                    IEnumerable<(string jobName, Run jobRun)> jobRunHistory = await databricksInfoProvider.GetFinishedJobRunsAsync(last, next);
+                    last = next;
+
+                    foreach ((string jobName, Run run) in jobRunHistory)
+                    {
+                        ReportJobRunAsMetric(jobName, run);
+                    }
+
+                    await Task.Delay(_options.Interval, stoppingToken);
+                }
+            }
+        }
+
+        private void ReportJobRunAsMetric(string jobName, Run run)
+        {
+            TextInfo text = new CultureInfo("en-US", useUserOverride: false).TextInfo;
+            Logger.LogInformation("Found finished job run with id {RunId}", run.RunId);
+
+            RunResultState resultState = run.State.ResultState ?? default(RunResultState);
```

Comment:
Why are we doing this? 🤔

Reply:
Bc it's being used somewhere else and so can't be `null`. 🤔

Reply:
So do we have to do this then?

Reply:
It wasn't done originally and maybe the response will always have this, but I still like to guard it like this. Call it healthy paranoia. 😅

Reply:
Let's add a guard for it instead indeed and make sure we don't use `default(RunResultState)` anymore

Reply:
Done.